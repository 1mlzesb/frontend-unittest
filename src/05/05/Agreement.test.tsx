import { render, screen } from "@testing-library/react";
import { Agreement } from "./Agreement";

test("fieldset의 접근 가능한 이름을 legend에서 인용한다", () => {
  render(<Agreement />);
  expect(
    screen.getByRole("group", { name: "이용계약 동의" })
  ).toBeInTheDocument();
});

test("체크박스가 체크되어 있지 않다", () => {
  render(<Agreement />);
  expect(screen.getByRole("checkbox")).not.toBeChecked();
});

test("이용계약 링크가 있다", () => {
  render(<Agreement />);
  expect(screen.getByRole("link")).toBeInTheDocument();
  expect(screen.getByRole("link")).toHaveTextContent("이용계약");
  expect(screen.getByRole("link")).toHaveAttribute("href", "/terms");
  expect(screen.getByRole("link", { name: "이용계약" })).toHaveAttribute(
    "href",
    "/terms"
  );
});

test("Snapshot: 이용계약 동의가 표시되어 있다", () => {
  const { container } = render(<Agreement />);
  expect(container).toMatchSnapshot();
});
