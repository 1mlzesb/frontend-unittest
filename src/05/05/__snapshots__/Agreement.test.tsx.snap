// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 이용계약 동의가 표시되어 있다 1`] = `
<div>
  <fieldset>
    <legend>
      이용계약 동의
    </legend>
    <label>
      <input
        type="checkbox"
      />
      서비스 
      <a
        href="/terms"
      >
        이용계약
      </a>
      을 확인했으며 이에 동의합니다
    </label>
  </fieldset>
</div>
`;
